### PR TITLE
JSON Spec Updates and SEQ JSON to EDSL fix

### LINKED ISSUE #668: Uploading generated seqjson uses old list notation
When uploading generated seqjson inside the Sequence Editor it generates the old list notation rather than the new object notation.

We may need to wait until we've updated to using the latest seqjson before addressing this ticket.

### PR DESCRIPTION
**Pull request by goetzrrGit**
* **Tickets addressed:** Closes #668
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description
Updating how we convert from eDSL to SeqJSON and vice-versa. There are new changes in the SPEC and this PR is using version 1.0.16 for the JSON SPEC. This is for the point release next week. There are tweaks to repeat arguments in 1.0.17 of the SPEC which @cohansen will be taking care of by bringing back arguments by position. 

## Verification
Ran the e2e test and all checks out. 

## Documentation
Probably some documentation needs to be updated. 

## Future work
Upgrading to SPEC 1.0.17

### CODE REVIEW

File: sequencing-server/package-lock.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "sequencing-server",
   "version": "1.1.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 3,
```
Comment (camargo) [inline sequencing-server/package-lock.json]: @goetzrrGit What version of Node/NPM are you using? The lock file should not have this many updates with the change you made to package.json. Make sure you are using node LTS (which comes with NPM 8.x). The lockfile version should remain on `2`.

**Comment by camargo** [comment]
Are there tickets associated with this PR or should we add this to the Project?

File: sequencing-server/package-lock.json (hunk 1 above)
Comment (goetzrrGit) [reply sequencing-server/package-lock.json]: This was generated by` gradlew assemble `which is node 18.13 using npm that is bundled in 18.13

**Comment by goetzrrGit** [comment]
@camargo  I accidentally hit `create PR` before updating the description with the ticket number. It should not link to an issue

**Review by cohansen — CHANGES_REQUESTED** [review]
Can you go through and comment some of the newly added code in the preface just to explain what it does?

File: sequencing-server/src/lib/codegen/CommandEDSLPreface.ts [hunk 2]
```diff
@@ -388,16 +388,120 @@ export class CommandStem<A extends Args[] | { [argName: string]: any } = [] | {}
...
+            .map(arg => {
+              return this.convertArgsToInterfaces(arg);
+            })
+            .flat(),
+          type: 'repeat',
+          name: key,
+        };
+        result.push(repeatArg);
+      } else {
+        if (typeof value === 'string') {
```
Comment (cohansen) [inline sequencing-server/src/lib/codegen/CommandEDSLPreface.ts]: Would this be better suited as a switch? Might be more clear.

**Comment by goetzrrGit** [comment]
@cohansen Sure I can do that

**Comment by camargo** [comment]
What would the [upgrade guide](https://nasa-ammos.github.io/aerie-docs/upgrade-guides/1-0-2-to-1-1-0/) look like for this?

**Comment by goetzrrGit** [comment]
@camargo you can add this to the upgrade guide.

The exported seqJSON now matches SPEC version 1.0.16, and a bug has been fixed in AERIE Sequence Editor that previously prevented users from transitioning from SeqJson to eDSL

**Review by cohansen — APPROVED** [review]
Looks good, thanks for adding those comments.

**Pull request merged**